src/Navigation.js

```diff
@@ -48,7 +48,7 @@ function _registerComponentNoRedux(screenID, generator) {
 
       render() {
         return (
-          <InternalComponent testID={screenID} navigator={this.navigator} {...this.state.internalProps} />
+          <InternalComponent testID={screenID} onPress={this.button ? this.button.onPress.bind(this.button) : null} navigator={this.navigator} {...this.state.internalProps} />
```

Comment:
Additionally, binding in props passed on render is generally frowned upon, since it happens every render. What do you think about lifting this out to a method on the subclass of `Screen` that is returned?